src/Domain/Stager.php

```diff
@@ -38,21 +40,43 @@ public function __construct(Filesystem $filesystem, ProcessFactory $processFacto
     }
 
     /**
-     * @param string[] $command
+     * @param string[] $composerCommand
+     *   The Composer command parts exactly as they would be typed in the
+     *   terminal. Example:
+     *
+     *   @code{.php}
+     *   $command = [
+     *     // "composer" is implied.
+     *     'require',
+     *     'lorem/ipsum:"^1 || ^2"',
+     *     '--with-all-dependencies',
+     *   ];
+     *   @endcode
      *
      * @throws \PhpTuf\ComposerStager\Exception\DirectoryNotFoundException
      * @throws \PhpTuf\ComposerStager\Exception\DirectoryNotWritableException
      * @throws \PhpTuf\ComposerStager\Exception\InvalidArgumentException
      * @throws \PhpTuf\ComposerStager\Exception\LogicException
      * @throws \Symfony\Component\Process\Exception\LogicException
      */
-    public function stage(array $command, string $stagingDir): void
+    public function stage(array $composerCommand, string $stagingDir): string
     {
-        $this->command = $command;
+        $this->composerCommand = $composerCommand;
         $this->stagingDir = $stagingDir;
+        $this->validate();
+        return $this->runCommand();
+    }
+
+    /**
+     * @throws \PhpTuf\ComposerStager\Exception\LogicException
+     * @throws \PhpTuf\ComposerStager\Exception\DirectoryNotFoundException
```

Comment:
As before, do we need `@throws` annotations for a method that doesn't actually directly throw anything?

Reply:
Ditto before. :)